src/librustc/mir/interpret/allocation.rs

```diff
@@ -183,7 +243,14 @@ impl<'tcx, Tag: Copy, Extra: AllocationExtra<Tag>> Allocation<Tag, Extra> {
         size: Size,
     ) -> InterpResult<'tcx, &[u8]>
     {
-        self.get_bytes_internal(cx, ptr, size, true)
+        self.get_bytes_internal(cx, ptr, size, true).map(|bytes_or_undef| match bytes_or_undef {
+            BytesOrElidedUndef::MaybeBytes(bytes) => bytes,
+            BytesOrElidedUndef::ElidedUndef => {
+                // Any other size would have triggered the definedness check.
+                debug_assert_eq!(size, Size::ZERO);
```

Comment:
Please make this `assert_eq!` unless we have benchmarks showing that this is expensive.

Reply:
It was @oli-obk who proposed a `debug_assert_eq!` [here](https://github.com/rust-lang/rust/pull/62655#discussion_r303239204).

Reply:
Oh lol I just noticed @oli-obk told you to use `debug_assert`.

@oli-obk I think we shouldn't elide this for release builds, this feels like the kind of thing that mis-use of the allocation API *anywhere in the Miri engine* can trigger.

Reply:
ok, I just assumed this to be a fairly hot path, but undef zst arrays are probably not in a hot path.